### PR TITLE
Add timeout parameter to rabbitmq module

### PR DESCRIPTION
**Pull request by prometheanfire**
##### SUMMARY
Add rabbitmq timeout option.

rabbitmqctl sometimes will hang, adding the timeout helps (and can be retried)

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
rabbitmq

##### ADDITIONAL INFORMATION
https://bugzilla.redhat.com/show_bug.cgi?id=1434593
https://groups.google.com/forum/#!topic/rabbitmq-users/iZWokxvhlaU

### CODE REVIEW

File: lib/ansible/modules/messaging/rabbitmq/rabbitmq_global_parameter.py [hunk 1]
```diff
@@ -85,6 +93,8 @@ def __init__(self, module, name, value, node):
     def _exec(self, args, run_in_check_mode=False):
         if not self.module.check_mode or (self.module.check_mode and run_in_check_mode):
             cmd = [self._rabbitmqctl, '-q', '-n', self.node]
+            if self.timeout:
```
Comment (resmo) [inline lib/ansible/modules/messaging/rabbitmq/rabbitmq_global_parameter.py]: not a blocker but timeout=0 would be omitted in this case (to whatever the default timeout is) i would use

~~~diff
-            if self.timeout:
+            if self.timeout is not None:
~~~

**Review by jgkirschbaum — APPROVED** [review]
Introducing the `timeout` parameter to `rabbitmqctl` seems to be useful for me. Why not implementing it in the other rabbitmq modules which use `rabbitmqctl`?

File: lib/ansible/modules/messaging/rabbitmq/rabbitmq_global_parameter.py (hunk 1 above)
Comment (prometheanfire) [reply lib/ansible/modules/messaging/rabbitmq/rabbitmq_global_parameter.py]: timeout=0 is still valid, though if it's useful is up for debate.

Comment (jgkirschbaum) [reply lib/ansible/modules/messaging/rabbitmq/rabbitmq_global_parameter.py]: `timeout=0` is valid, but it's not entering the `if` branch if the `is not None` is missing.

Comment (prometheanfire) [reply lib/ansible/modules/messaging/rabbitmq/rabbitmq_global_parameter.py]: updated it, thanks

File: lib/ansible/modules/messaging/rabbitmq/rabbitmq_global_parameter.py [hunk 2]
```diff
@@ -44,14 +44,21 @@
     required: false
     default: present
     choices: [ 'present', 'absent']
+  timeout:
+    description:
+      - Specify if a timeout is desired
+    required: false
+    default: null
```
Comment (Akasurde) [inline lib/ansible/modules/messaging/rabbitmq/rabbitmq_global_parameter.py:51]:
```suggestion
    default: null
    version_added: '2.10'
```

**Pull request closed without merging**